accepted/2025/rid-specific-tool-packages.md

```diff
@@ -0,0 +1,91 @@
+
+The RID-specific tool packages should be named using the convention `<Toolname>.<RID>`.  For example, if the `dotnetsay` tool has RID-specific packages, they would be named `dotnetsay.win-x64`, `dotnetsay.linux-x64`, etc.
+
+The tool manifest for the primary package would look like this:
+
+```xml
+<DotNetCliTool Version="1">
+  <Commands>
+    <Command Name="dotnet-say" />
+  </Commands>
+  <RuntimeIdentifierPackages>
+    <RuntimeIdentifierPackage RuntimeIdentifier="win-x64" Id="dotnet-say.win-x64" Version="1.0.0" />
+    <RuntimeIdentifierPackage RuntimeIdentifier="linux-x64" Id="dotnet-say.linux-x64" Version="1.0.0" />
+  </RuntimeIdentifierPackages>
+</DotNetCliTool>
+```
+
+Note that the Command has no EntryPoint or Runner, as that is defined in the RID-specific package.  The command name is still included because that may be useful for tooling.
+
+## Entry points
+
+Recall that the tool manifest file has a section that looks like this:
+
+```xml
+<Command Name="dotnet-say" EntryPoint="dotnet-say.dll" Runner="dotnet" />
+```
+
+Currently, the only supported value for `Runner` is `dotnet`, which means that the `EntryPoint` is a DLL that can be launched with the `dotnet` executable.  When a global tool is installed, the .NET SDK will create a version of the .NET apphost (called a shim) in the global tools folder.  This shim has a filename matching the tool command name so that the tool can be launched from the path.  The shim also has the relative path to the tool entry point embedded in it so that it can launch the correct tool.  Tools can also include their own shims so that they can be properly signed.
+
+To support NativeAOT .NET apps (or non .NET apps), we should support an additional `executable` runner for executables that can be launched natively by the operating system.  On Unix-like operating systems, we can create a symlink in the global tools folder to the tool entry point executable.  On Windows, we may create a batch file that launches the entry point executable.  If this has drawbacks we might create a new type of shim/launcher.
+
+An alternative would be to copy the whole entry point executable to the global tools folder.  However, there might be other files in the same folder that the tool depends on, so we don't currently plan to do this.
```

Comment:
> However, there might be other files in the same folder that the tool depends on, so we don't currently plan to do this.

1. A tool using [ClangSharp](https://github.com/dotnet/ClangSharp) would be an excellent test case for a rid-specific tool package, because the native clang dependency is _big_.
2. A tool using ClangSharp would have a dependency on a native dynamic library (dll/dylib), and as such copying just the executable would not work.